pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -86,6 +88,14 @@ func (ku *KubeUtil) init() error {
 		}
 	}
 
+	if pkgconfigsetup.Datadog().GetBool("kubelet_use_api_server") {
+		ku.useAPIServer = true
+		ku.kubeletClient.config.nodeName, err = ku.GetNodename(context.Background())
+		if err != nil {
+			return err
```

Comment:
Question: if we cannot detect the nodename, would we not want `kubeutil` to init at all (i.e. will everything else called after fail?)

Reply:
discussed offline: in the case that the nodename can't be detected, hostname resolution will fail which will cause the agent to not startup anyway